src/core_plugins/kibana/public/home/sample_data_client.js

```diff
@@ -18,30 +18,36 @@
  */
 
 import { kfetch } from 'ui/kfetch';
+import chrome from 'ui/chrome';
+import { indexPatternService } from './kibana_services';
 
 const sampleDataUrl = '/api/sample_data';
 
+function clearIndexPatternsCache() {
+  indexPatternService.getIds.clearCache();
+}
+
 export async function listSampleDataSets() {
   return await kfetch({ method: 'GET', pathname: sampleDataUrl });
 }
 
-export async function installSampleDataSet(id, defaultIndex, getConfig, setConfig, clearIndexPatternsCache) {
+export async function installSampleDataSet(id, defaultIndex) {
   await kfetch({ method: 'POST', pathname: `${sampleDataUrl}/${id}` });
 
-  const existingDefaultIndex = await getConfig('defaultIndex');
+  const existingDefaultIndex = chrome.getUiSettingsClient().get('defaultIndex');
```

Comment:
I think using the `getUiSettingsClient().isDefault('defaultIndex')` method here would be a bit nicer to check, if that setting hasn't yet been changed by the user.